### DISCUSSION TITLE
Python setuptools pip installer

### DISCUSSION THREAD

**Discussion by MarkOliver1959** [mathoudebine/turing-smart-screen-python#378] [Ideas]
I've created a pip installer and lots of bits of code I had to change to make it work.

This zip file must not to be blindly merged with any of the git repository it is purely here to allow review. I would have created a fork with the root directory moved. That shuffle seemed difficult and could have nasty hard to resolve problems.  If you want to merge then use Beyond Compare or WinMerge.

I'm only dropping in the change files and and scripts for Matthieu to review. 

[tsmart.setup-bits.zip](https://github.com/mathoudebine/turing-smart-screen-python/files/13212285/tsmart.setup-bits.zip)

**Comment by MarkOliver1959** [comment]
Tried to upload the python wheel file and it's was over the 25 MB limit :-( 
So I removed the --Theme examples subdirectory that was buried under the themes. Hopefully this will install.  
 
[Readme-Installer.md](https://github.com/mathoudebine/turing-smart-screen-python/files/13212471/Readme-Installer.md)
[Turing_Smart_Screen_Python.whl.zip](https://github.com/mathoudebine/turing-smart-screen-python/files/13216683/Turing_Smart_Screen_Python.whl.zip)

**Comment by mathoudebine** [comment]
Thanks for sharing! Nice work, it seems you went pretty far into the investigation!
If I have some time, I will try to integrate it to the main repository
It would certainly be better for people to download/install/update a single file instead of the whole Git repository.
I agree the themes should not be part of the Wheel file, they should be packaged elsewhere (in a .zip file?)

**Reply by MarkOliver1959** [reply]
In that **whl** file I attached above, I bundled the main working Themes, but there was another huge directory "--Theme examples" had lots and lots of backgrounds and bits for themes, which I had to leave out   That is best left as a github link in a readme.md so people can grab that separately. 

At some point I was hoping to integrate that menu.sh functionality into the configure.py so you Tab over to a configuration page that did that service generation / configuration within the configure.py rather than that menu.  The only tricky bit is how to get Sudo functionality in the same simplistic manor that Bash does it.  
Then do the same for the Windows  version.

In regards to integrating - The main issues is stupid Python and setuptools conventions of having project root directory above your project.  It's that whole package v/s script stupidity with __init_.py and paths - aaarrrggghhh  !!!!
